api/task.proto

```diff
@@ -0,0 +1,88 @@
+// Copyright 2018 The OpenPitrix Authors. All rights reserved.
+// Use of this source code is governed by a Apache license
+// that can be found in the LICENSE file.
+
+syntax = "proto3";
+package openpitrix;
+
+// set go package name to pb
+option go_package = "pb";
+
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/timestamp.proto";
+import "google/api/annotations.proto";
+import "protoc-gen-swagger/options/annotations.proto";
+
+option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
+	info: {
+		title: "OpenPireix Project";
+		version: "0.0.1";
+		contact: {
+			name: "OpenPireix Project";
+			url: "https://openpitrix.io";
+		};
+	};
+	schemes: HTTP;
+	consumes: "application/json";
+	produces: "application/json";
+};
+
+
+message CreateTaskRequest {
+	google.protobuf.StringValue _ = 1; // placeholder for task_id
+	google.protobuf.StringValue job_id = 2;
+	google.protobuf.StringValue task_action = 3;
+	google.protobuf.StringValue directive = 4;
+}
+
+message CreateTaskResponse {
+	google.protobuf.StringValue task_id = 1;
+	google.protobuf.StringValue job_id = 2;
+}
+
+message Task {
+	google.protobuf.StringValue task_id = 1;
+	google.protobuf.StringValue job_id = 2;
+	google.protobuf.StringValue task_action = 3;
+	google.protobuf.StringValue status = 4;
+	google.protobuf.UInt32Value error_code = 5;
+	google.protobuf.StringValue directive = 6;
+	google.protobuf.StringValue executor = 7;
+	google.protobuf.StringValue owner = 8;
+	google.protobuf.StringValue target = 9;
+	google.protobuf.StringValue node_id = 10;
+	google.protobuf.Timestamp create_time = 11;
+	google.protobuf.Timestamp status_time = 12;
+}
+message DescribeTasksRequest {
+	repeated string task_id = 1;
+	repeated string job_id = 2;
+	repeated string status = 3;
+	uint32 limit = 4; // uint64 marshal to json will convert to string
+	uint32 offset = 5;
+}
+message DescribeTasksResponse {
+	uint32 total_count = 1;
+	uint32 limit = 2;
```

Comment:
`limit` is not necessary in response.

Reply:
When user's limit is bigger than 200, then he/she should know the limit.

Reply:
There is no need to define in the Response, we can define it as a comment in Request. Such as:
```proto
message DescribeTasksRequest {
    // default is 20, max value is 200
    uint32 limit = 1;
    // default is 0
    uint32 offset = 2;
}
```
swagger-go will convert this comment to arguments' description.

Reply:
if no limit in response, it should return error if `limit > 200` in req

Reply:
OK, I see we not return limit in qingcloud.

Reply:
@chai2010  No need to return error, just return ower largest limit record is ok.